File: VB/CustomActionControl.Module.Win/ActionControls/BarCheckItemCheckableSimpleActionControl.vb
Imports System
Imports DevExpress.ExpressApp.Win.Templates.Bars.ActionControls
Imports DevExpress.XtraBars

Namespace CustomActionControl.Module.Win.ActionControls
    Public Class BarCheckItemCheckableSimpleActionControl
        Inherits BarItemActionControl(Of BarCheckItem)

        Private Sub BarItem_ItemClick(ByVal sender As Object, ByVal args As ItemClickEventArgs)
            RaiseExecute()
        End Sub
        Protected Overrides Sub OnEndInit()
            MyBase.OnEndInit()
            AddHandler BarItem.ItemClick, AddressOf BarItem_ItemClick
        End Sub
        Protected Sub RaiseExecute()
            If IsConfirmed() Then
                RaiseEvent Execute(Me, EventArgs.Empty)
            End If
        End Sub
        Public Sub New()
        End Sub
        Public Sub New(ByVal actionId As String, ByVal item As BarCheckItem)
            MyBase.New(actionId, item)
        End Sub
        Public Property Checked() As Boolean
            Get
                Return BarItem.Checked
            End Get
            Set(ByVal value As Boolean)
                BarItem.Checked = value
            End Set
        End Property
        Public Event Execute As EventHandler(Of EventArgs)
    End Class
End Namespace

File: CS/EFCore/CustomActionEF/CustomActionEF.Win/ActionControls/BarCheckItemCheckableSimpleActionControl.cs
using System;
using DevExpress.ExpressApp.Win.Templates.Bars.ActionControls;
using DevExpress.XtraBars;

namespace CustomActionControl.Module.Win.ActionControls {
    public class BarCheckItemCheckableSimpleActionControl : BarItemActionControl<BarCheckItem> {
        private void BarItem_ItemClick(object sender, ItemClickEventArgs args) {
            RaiseExecute();
        }
        protected override void OnEndInit() {
            base.OnEndInit();
            BarItem.ItemClick += BarItem_ItemClick;
        }
        protected void RaiseExecute() {
            if(IsConfirmed() && Execute != null) {
                Execute(this, EventArgs.Empty);
            }
        }
        public BarCheckItemCheckableSimpleActionControl() { }
        public BarCheckItemCheckableSimpleActionControl(string actionId, BarCheckItem item) : base(actionId, item) { }
        public bool Checked {
            get { return BarItem.Checked; }
            set { BarItem.Checked = value; }
        }
        public event EventHandler<EventArgs> Execute;
    }
}
